rfd/0151-database-permission-management.md

```diff
@@ -0,0 +1,421 @@
+the database schema will be read and passed through the import rules. Each
+import rule defines a set of database labels that must match in order for the
+import rule to fire. The rules are processed in order (as defined by the
+`priority` field).
+
+Each import rule defines a set mappings. Individual mapping consists of a set of
+attributes to match, like `name`, `object kind` or `schema`, and a list of
+labels to apply. The expected values for an attribute are specified as a list,
+while the matching uses glob-like semantics, with `*` matching any set of
+characters.
+
+#### Examples
+
+The `widget-prod` import rule applies to databases with `env: prod` label. It
+has the priority 10. If there is a database object that matches the clauses in
+`spec.mappings.match`, the import rule will apply the specified labels to the
+database object (`env: prod`, `product: WidgetMaster3000`).
+
+```yaml
+kind: db_object_import_rule
+version: v1
+metadata:
+  name: rule_widget_prod
+spec:
+  priority: 10
+  db_labels:
+    env: prod
+  mappings:
+    - match:
+        databases:
+          - 'Widget*'
+        db_service_names:
+          - all-things-widget
+        names:
+          - '*sales*'
+        object_kinds:
+          - table
+          - view
+          - procedure
+        protocol_names:
+          - postgres
+        schemas:
+          - widget
+          - sales
+          - public
+          - secret
+      add_labels:
+        env: prod
+        product: WidgetMaster3000
+```
+
+There is a Postgres database with a matching labels. Parsing the schema, a
+number of objects is found, including the following one which matches the import
+rule above (pattern matches are noted):
+
+```yaml
+kind: db_object
+version: v1
+metadata:
+  name: widget-sales
+spec:
+  database: WidgetUltimate # matches 'Widget*'
+  db_service_name: all-things-widget
+  name: widget-sales # matches '*sales*'
+  object_kind: table
+  protocol: postgres
+  schema: sales
+```
+
+After processing, the import rule has applied the labels to the object. As there
+are no other import rules matching this object, it is stored in the backend in
+this form.
+
+```yaml
+kind: db_object
+version: v1
+metadata:
+  name: widget-sales
+  labels:
+    env: prod
+    product: WidgetMaster3000
+spec:
+  database: WidgetUltimate
+  db_service_name: all-things-widget
+  name: widget-sales
+  object_kind: table
+  protocol: postgres
+  schema: sales
+```
+
+Any particular attribute can be omitted from the import rule; an import rule
+with empty `match` part will match all objects.
+
+A single rule can also specify multiple sets of labels to be applied. This is
+useful for applying a label with different set of values, depending on database
+object attributes.
+
+```yaml
+kind: db_object_import_rule
+version: v1
+metadata:
+  name: mark-confidential
+spec:
+  priority: 20
+  db_labels:
+    env: prod
+  mappings:
+    - match:
+        schemas:
```

Comment:
I think this (and other `db_object_import_rule` examples) needs updating to move out schemas to a separate "scopes" field to match what's in your demo?

Also, what would be supported in "scopes"? Just schemas for now?

Reply:
`schema_names` and `database_names`, per https://github.com/gravitational/teleport/pull/33734#discussion_r1433683529.

Reply:
Examples updated, thanks.